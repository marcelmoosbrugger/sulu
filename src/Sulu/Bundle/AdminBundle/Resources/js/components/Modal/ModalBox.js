// @flow
import React from 'react';
import Icon from '../Icon';
import type {ModalProps} from './types';
import Actions from './Actions';
import modalBoxStyles from './modalBox.scss';

type Props = ModalProps & {
    onRequestClose: () => void,
}

const CLOSE_ICON = 'times';

export default class ModalBox extends React.PureComponent<Props> {
    static defaultProps = {
        actions: [],
    };

    render() {
        const {title, onRequestClose, children, actions, onConfirm, confirmText} = this.props;

        return (
            <section className={modalBoxStyles.box}>
                <header>
                    {title}
                    <Icon name={CLOSE_ICON} className={modalBoxStyles.icon} onClick={onRequestClose} />
                </header>
                <article>{children}</article>
                <footer>
                    <Actions actions={actions} />
                    <button className={modalBoxStyles.confirmButton} onClick={onConfirm}>{confirmText}</button>
                </footer>
            </section>
        );
    }
}
