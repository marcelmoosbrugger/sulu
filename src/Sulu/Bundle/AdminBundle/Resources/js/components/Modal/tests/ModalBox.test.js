/* eslint-disable flowtype/require-valid-file-annotation */
import React from 'react';
import {render, shallow} from 'enzyme';
import ModalBox from '../ModalBox';

test('The component should render', () => {
    const onRequestClose = () => {};
    const onConfirm = () => {};
    const actions = [
        {title: 'Action 1', onClick: () => {}},
        {title: 'Action 2', onClick: () => {}},
    ];
    const box = render(
        <ModalBox
            title="My title"
            actions={actions}
            onRequestClose={onRequestClose}
            onConfirm={onConfirm}
            confirmText="Alright mate!" >
            <p>My modal content</p>
        </ModalBox>
    );
    expect(box).toMatchSnapshot();
});

test('The component should request to be closed when the close icon is clicked', () => {
    const onRequestClose = jest.fn();
    const onConfirm = () => {};
    const box = shallow(
        <ModalBox
            title="My title"
            onRequestClose={onRequestClose}
            onConfirm={onConfirm}
            confirmText="Alright mate!" >
            <p>My modal content</p>
        </ModalBox>
    );
    box.find('Icon').props().onClick();
    expect(onRequestClose).toBeCalled();
});

test('The component should call the callback when the confirm button is clicked', () => {
    const onRequestClose = () => {};
    const onConfirm = jest.fn();
    const box = shallow(
        <ModalBox
            title="My title"
            onRequestClose={onRequestClose}
            onConfirm={onConfirm}
            confirmText="Alright mate!" >
            <p>My modal content</p>
        </ModalBox>
    );
    box.find('.confirmButton').simulate('click');
    expect(onConfirm).toBeCalled();
});
