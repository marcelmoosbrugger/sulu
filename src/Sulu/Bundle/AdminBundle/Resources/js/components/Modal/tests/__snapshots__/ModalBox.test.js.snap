// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render 1`] = `
<section
  class="box"
>
  <header>
    My title
    <span
      aria-hidden="true"
      class="icon fa fa-times"
    />
  </header>
  <article>
    <p>
      My modal content
    </p>
  </article>
  <footer>
    <div
      class="actions"
    >
      <button
        class="action"
      >
        Action 1
      </button>
      <button
        class="action"
      >
        Action 2
      </button>
    </div>
    <button
      class="confirmButton"
    >
      Alright mate!
    </button>
  </footer>
</section>
`;
